prov/gni/src/gnix_ep.c

```diff
@@ -576,6 +588,8 @@ int gnix_ep_open(struct fid_domain *domain, struct fi_info *info,
 		}
 	}
 
-	if (nic != NULL) free(nic);
+	if (nic != NULL)
+		free(nic);
+	pthread_mutex_unlock(&gnix_nic_list_lock);
```

Comment:
Yeah, I just felt like it was a rather large critical section. Not sure if it actually will matter though...